below/procfs/src/lib.rs

```diff
@@ -1305,21 +1309,35 @@ impl NetReader {
     }
 
     pub fn read_netstat(&self) -> Result<NetStat> {
-        let netstat_map = self.read_kv_diff_line("netstat")?;
-        let snmp_map = self.read_kv_diff_line("snmp")?;
-        let snmp6_map = self.read_kv_same_line("snmp6")?;
+        let netstat_map = handle_io_error(&self.logger, self.read_kv_diff_line("netstat"))?;
```

Comment:
Can you add a comment before this block for the reasons why it's valid for files to be missing? (e.g. the reasons Daniel mentioned below)